src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/HtmlCSharp/DefaultLSPRequestInvoker.cs

```diff
@@ -75,7 +77,17 @@ public DefaultLSPRequestInvoker(ILanguageClientBroker languageClientBroker)
 
             var (_, resultToken) = await task.ConfigureAwait(false);
 
-            var result = resultToken != null ? resultToken.ToObject<TOut>() : default;
+            // We need these converters so we don't lose information as part of the deserialization.
+            var serializer = new JsonSerializer();
```

Comment:
Sorry this is late. Can we do this once instead of every invocation?

Reply:
https://github.com/dotnet/aspnetcore-tooling/pull/1839